Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -265,7 +267,7 @@ - (id)convertedServerTimestamp:(const FieldValue &)value
 }
 
 - (id)convertedReference:(const FieldValue &)value {
-  const auto &ref = value.reference_value();
+  auto ref = value.reference_value();
```

Comment:
Question: what's the reason for this change?

Reply:
This is a mistake. It has leaked forward from a change that made `reference_value` return a value.

Reverted.